### DISCUSSION TITLE
[Service Fabric] Autoscaling Policies

### DISCUSSION THREAD

**Discussion by meanin** [AzureCommunityPL/group#37] [Pomoc] [answered]
Cześć,
próbuję skonfigurować polityki do autoskalowania na naszym klastrze Service Fabric. Korzystam z [microsoft docs](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-cluster-resource-manager-autoscaling). Póki co jestem w stanie ustawić działającą pojedynczą politykę działającą na standardowych metrykach (memory in MB albo CPU usage). Chciałbym ustawić dwie polityki:
```   
   <Service Name="SomeApp.SomeServiceName" ServicePackageActivationMode="ExclusiveProcess">
      <StatelessService ServiceTypeName="SomeApp.SomeServiceNameType" InstanceCount="1">
         <SingletonPartition />
         <ServiceScalingPolicies>
            <ScalingPolicy>
               <AveragePartitionLoadScalingTrigger MetricName="servicefabric:/_CpuCores" LowerLoadThreshold="0.2" UpperLoadThreshold="0.3" ScaleIntervalInSeconds="10" />
               <InstanceCountScalingMechanism MinInstanceCount="1" MaxInstanceCount="5" ScaleIncrement="1" />
            </ScalingPolicy>
            <ScalingPolicy>
               <AveragePartitionLoadScalingTrigger MetricName="servicefabric:/_MemoryInMB" LowerLoadThreshold="150" UpperLoadThreshold="200" ScaleIntervalInSeconds="10" />
               <InstanceCountScalingMechanism MinInstanceCount="1" MaxInstanceCount="5" ScaleIncrement="1" />
            </ScalingPolicy>
         </ServiceScalingPolicies>
      </StatelessService>
   </Service>
```

Czy jest szansa, że mam tu błąd? Czy w ogóle założenie wielu policy dla jednego service'u jest błędne?

**Comment by kamil-mrzyglod** [comment, marked as answer]
Myślę, że masz to wyjaśnione tutaj - https://github.com/MicrosoftDocs/azure-docs/issues/11632

Dalej masz to w dokumentacji - https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-cluster-resource-manager-autoscaling, cytując:

> Currently there is support for only one scaling policy per service, and only one scaling trigger per scaling policy.

Więc niestety nie jest to obsługiwane.
